src/base/dev/vga/vgaemu.c

```diff
@@ -805,15 +803,19 @@ unsigned vga_read_dword(dosaddr_t addr)
 void vga_mark_dirty(dosaddr_t s_addr, int len)
 {
   unsigned vga_page, abeg, aend, addr;
+  int i, j;
   abeg = s_addr & PAGE_MASK;
   aend = (s_addr + len - 1) & PAGE_MASK;
   for (addr = abeg; addr <= aend; addr += PAGE_SIZE) {
-    if (!vga_write_access(addr))
```

Comment:
This was the UMB check, right?
Is this really faster to skip it?
Maybe instead you can put it into
a header and inline?

Reply:
This is just replaced by the check between `vga.mem[i].base_page` and `vga.mem[i].base_page+pages`, outside that range `_vga_emu_adjust_protection` won't do anything so the `vga_write_access` became superfluous.

Reply:
argh the only trouble with this check is that umb_b0=1 introduces a hole. Maybe I should increase VGAEMU_MAX_MAPPINGS by 1 and add in a special mapping for 0xb8000 to account for the possible hole, so the first mapping goes from a0000-b8000, a0000-b0000 or b0000-b8000, or empty depending on the umb settings.

Reply:
the second bank mapping is then always from b8000 to c0000.